File: Java/Array/Sum.java
import java.util.Scanner;

public class Array {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int[] arr = new int[n];

		for (int i = 0; i < n; i++) {
			arr[i] = sc.nextInt();
		}
		//Tong cua so hang dau va cuoi cua chuoi
        System.out.println("Sum of the first and last elements: " + (arr[0] + arr[n - 1]));
		
		//Tong cua cac so hang trong chuoi
        int sum = 0;
        for (int i = 0; i < n; i++) {
            sum += arr[i];
        }
        System.out.println("Sum of all elements: " + sum);
        
        //Dao nguoc cac so hang trong chuoi
        System.out.print("Array in reverse order: ");
        for (int i = n - 1; i >= 0; i--) {
            System.out.print(arr[i] + " ");
        }
	}
}

File: C/Array/Sum.c
#include <stdio.h>

int main() {
    int a[1000], n;
    scanf("%d", &n);

    // Validate the size of the array
    if (n <= 0 || n > 1000) {
        return 1;
    }

    int sum = 0;
    for (int i = 0; i < n; i++) {
        printf("a[%d] = ", i);
        scanf("%d", &a[i]);
        sum += a[i];
    }

    printf("Sum of all elements: %d\n", sum);
    printf("Sum of the first and last elements: %d\n", a[0] + a[n - 1]);
    printf("Array in reverse order: ");
    for (int i = n - 1; i >= 0; i--) {
        printf("%d ", a[i]);
    }
    return 0;
}
